### ISSUE TITLE
Set up production instance for mozfest-event-app

### ISSUE THREAD

**Issue by mmmavis** [MozillaFoundation/mofo-devops#415]
### Details

As discussed in https://github.com/mozilla/mozfest-event-app/issues/248#issuecomment-247022243 , we should

1. [x] TODO No.1 ) make sure pushing to `gh-pages` triggers updates to https://app.mozillafestival.org 
2. [x] TODO No.2 ) config mozbot so that non devs can easily push things from staging to production
3. [x] TODO No.3 ) config mozbot so that we can easily trigger script(@mmmavis to work on this script) used for updating session data (event app spreadsheet -> format data into json -> commit .json file to mozfest-event-app's `master` branch) (I will chat with Yousef on this, as I'm having trouble recalling how we did it for the all hands app. Via some community owned Jenkin?)
4. [ ] ~~TODO No.4 ) schedule app data should be updated every 15 mins.~~

Once the above tasks are completed, the workflow to get updated session data onto https://app.mozillafestival.org will be

1. staff / MozFest organizers update data on the event app spreadsheet
2. staff / MozFest organizers trigger mozbot command (todo no.3) to get updated data committed to `master` branch (this then triggers updates to staging instance)
3. staff / MozFest organizers verify changes on staging are good
4. staff / MozFest organizers trigger mozbot command (todo no.2) to get staging -> production

@cadecairos does ^ sound sane to you?
 

### Info & links

- https://github.com/mozilla/mozfest-event-app/issues/248

**Comment by mmmavis** [comment]
Hey @cadecairos

### For now

Production site: https://app.mozillafestival.org/ currently shows last year's schedule. We would like to update it with some sort of "2016 Schedule is coming soon" message. What's the best way to do it? Will pushing my updates to the `backup-2015-mozfest-app` branch do the trick? https://github.com/mozilla/mozfest-event-app/tree/backup-2015-mozfest-app

### On or before **Oct 17** 

We would like to refresh the production entirely with our 2016 app. This means the no.1 TODO item needs to be done before then

>  TODO No.1 ) make sure pushing to gh-pages triggers updates to https://app.mozillafestival.org

I think we don't have to worry about other Mozbot related todos before Oct 17 as I can do manually push every once in awhile. (Mozbot helpers **do need** to be in place few days before MozFest.)

**Comment by cadecairos** [comment]
@mmmavis 

> Production site: https://app.mozillafestival.org/ currently shows last year's schedule. We would like to update it with some sort of "2016 Schedule is coming soon" message. What's the best way to do it? Will pushing my updates to the backup-2015-mozfest-app branch do the trick? https://github.com/mozilla/mozfest-event-app/tree/backup-2015-mozfest-app

If you make changes to that branch, and update the "on" condition in `.travis.yml` so that the repo and branch are "mozilla/mozfest-event-app" and "backup-2015-mozfest-app" respectively, it should deploy your updates to the correct s3 bucket that's serving the current content you see on app.mozillafestival.org

Once we're ready to push the new app to production, we can grab the deploy section of `.travis.yml` and update it to deploy to production when something is merged into `gh-pages`

**Comment by cadecairos** [comment]
> grab the deploy section of `.travis.yml`

(in the backup-2015-mozfest-app branch)

**Comment by mmmavis** [comment]
Sounds great! I'll let the MozFest team know. 🍻

**Comment by cadecairos** [comment]
@mmmavis what's the timeline for getting this live?

**Comment by mmmavis** [comment]
hey @cadecairos , the program team is gonna share production app to public next Monday (or Tuesday I forgot). I will update the production instance with the new app this Friday PST morning .

Mozbot related tasks don't have to be done this week as I can still update the app via regular git push until mozbot is ready.

**Comment by mmmavis** [comment]
Note: I added a 4th todo to https://github.com/MozillaFoundation/mofo-devops/issues/415#issue-177911955 . Again, it doesn't have to be done this week.

**Comment by cadecairos** [comment]
> No.4 ) schedule app data should be updated every 15 mins.

we can add this as a task in https://github.com/MozillaFoundation/mofo-cron

**Comment by mmmavis** [comment]
Hello hello @flamingspaz 😃 

It's about that time of the year again! MozFesttt! 🎉 

I tried to log into https://jenkins.services.mozilla.community/login?from=/  but I think my creds don't work anymore. Mind give a hand? I would like to set up mozbot like what you did for Mozilla's London All-hands App this year. If you can grant me access to the community Jenkin I can try setting it up myself (but expect questions from me, heh).

**Comment by flamingspaz** [comment]
heya @mmmavis!

We had some issues with the storage on our old Jenkins instance and lost the users unfortunately, but I've just sent @cadecairos an email with creds he can fwd to you. You should see the template app and you can copy it into the new one.

I'm not sure if https://github.com/mozilla/schedule-app-data-processor/ is the correct repo but if it is, then the cryptography package will need to be updated as the version specified doesn't work with newer OpenSSL versions.

**Comment by mmmavis** [comment]
Awesome, thanks @flamingspaz ! ✨ 

This is the repo for the MozFest app data processor: https://github.com/mozilla/mozfest-event-app-data-processor/blob/master/requirements.txt . Does updating `cryptography` to [v1.5.2](https://pypi.python.org/pypi/cryptography/) sound good?

@cadecairos do you mind forwarding me the creds I need to log in to the community owned Jenkin instance? 😃

**Comment by flamingspaz** [comment]
yup, 1.5.2 should work fine. I opened a PR to update the ones that were failing, and it seems to work now.

**Comment by mmmavis** [comment]
### Updates:

After discussing with the team we decided to just have 2 mozbot commands (no cron job)

1. One command to trigger the Python script (Python script gets data from Google Spreadsheet, formats it, and commits `sessions.json` to the` master` branch. In short, it updates the `staging` instance )
2. One command to merge `master` into `gh-pages` (which updates production site)

I'm working on the Jenkin job to update staging and @cadecairos is doing staging -> production Jenkin job.

**Comment by mmmavis** [comment]
@cadecairos Jenkin job to trigger updates to `master` (staging) is done! 🎉 It's called `mofo-deploy-mozfest-app-staging` in case you need to link it to mozbot 😉

**Comment by cadecairos** [comment]
@mmmavis once we confirm the jenkins jobs work, can you close this out?

**Comment by mmmavis** [comment]
@cadecairos it didn't work for me still. Please see error messages in `mozfest` channel in Chat.

```
mozbot
9:01 PM
@mavis Jenkins says: Status 403 <html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Error 403 No valid crumb was included in the request</title>
</head>
<body><h2>HTTP ERROR 403</h2>
<p>Problem accessing /job/mofo-deploy-mozfest-app-staging/build. Reason:
<pre>    No valid crumb was included in the request</pre></p><hr><i><small>Powered by Jetty://</small></i><hr/>
</body>
</html>
```

**Comment by cadecairos** [comment]
I think we've worked out all the kinks in jenkins.

🚀 

This is resolved

**Comment by mmmavis** [comment]
Y
E
A
H
!!!
✨
